### PR TITLE
Ngoc dpd

### PR DESCRIPTION
**Pull request by DUYNGOC97**
1. Implement product api BE
2. Implement category api BE

### CODE REVIEW

File: src/main/java/com/codegym/aurora/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ protected SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exce
                 .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
 
         httpSecurity.authorizeHttpRequests() // links start with /api/
-                .antMatchers("/api/*", "/api/login","api/register-user", "api/register-admin") // perform segregate authorize
+                .antMatchers("/api/home", "/api/login","api/register") // perform segregate authorize
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/configuration/SecurityConfiguration.java]: api endpoint doesn't match with controller

**Review by mainhuitvt96 — APPROVED** [review]

File: src/main/java/com/codegym/aurora/controller/store_front/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/products")
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/controller/store_front/CategoryController.java]: invalid end-point name

File: src/main/java/com/codegym/aurora/controller/store_front/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            @RequestParam(defaultValue = "10") int size) {
+        Page<HomeProductResponseDTO> homeProductResponseDTOS = productService.getProductsPage(page, size);
+        if (homeProductResponseDTOS.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        } else {
+            return ResponseEntity.ok(homeProductResponseDTOS);
+        }
+    }
+
+    @GetMapping("product/{productId}")
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/controller/store_front/ProductController.java]: invalid endpoint

File: src/main/java/com/codegym/aurora/entity/Category.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @Column(name = "NAME", nullable = false)
+    private String name;
+
+    @Column(name = "IS_DELETE", nullable = false)
+    private boolean idDelete;
+
+    @Column(name = "IS_ACTIVATED", nullable = false)
+    private boolean isActivated;
+
+    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/entity/Category.java]: use merge/persist cascade type

File: src/main/java/com/codegym/aurora/entity/Product.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    @Column(name = "IMAGE_URL", nullable = false)
+    private String imageUrl;
+
+    @Column(name = "IS_DELETE", nullable = false)
+    private boolean isDelete;
+
+    @Column(name = "IS_ACTIVATED", nullable = false)
+    private boolean isActivated;
+
+    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/entity/Product.java]: use  merge/persist cascade type

File: src/main/java/com/codegym/aurora/entity/ProductImage.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package com.codegym.aurora.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/entity/ProductImage.java]: use specifically import

File: src/main/java/com/codegym/aurora/entity/SubCategory.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package com.codegym.aurora.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/entity/SubCategory.java]: use specifically import

File: src/main/java/com/codegym/aurora/payload/response/ProductDetailResponseDTO.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+@AllArgsConstructor
+public class ProductDetailResponseDTO {
+
+    private long id;
+
+    private String name;
+
+    private long price;
+
+    private int weighT;
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/payload/response/ProductDetailResponseDTO.java:22]: please follow java naming rules

File: src/main/java/com/codegym/aurora/payload/response/ProductImageResponseDTO.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.codegym.aurora.payload.response;
+
+import lombok.*;
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/payload/response/ProductImageResponseDTO.java]: use specifically import

File: src/main/java/com/codegym/aurora/repository/UserRepository.java [hunk 10]
```diff
@@ -7,4 +7,4 @@ public interface UserRepository extends JpaRepository<User, Integer> {
     User findByUsername(String username);
 
     User findUserById(long id);
```
Comment (ChungNguyenVr912) [inline src/main/java/com/codegym/aurora/repository/UserRepository.java]: please check this method!

**Review by ChungNguyenVr912 — CHANGES_REQUESTED** [review]
please fix all issues

File: src/main/java/com/codegym/aurora/configuration/SecurityConfiguration.java (hunk 1 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/configuration/SecurityConfiguration.java]: FIXED

File: src/main/java/com/codegym/aurora/controller/store_front/CategoryController.java (hunk 2 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/controller/store_front/CategoryController.java]: FIXED

File: src/main/java/com/codegym/aurora/controller/store_front/ProductController.java (hunk 3 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/controller/store_front/ProductController.java]: FIXED

File: src/main/java/com/codegym/aurora/entity/Category.java (hunk 4 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/entity/Category.java]: FIXED

File: src/main/java/com/codegym/aurora/entity/Product.java (hunk 5 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/entity/Product.java]: FIXED

File: src/main/java/com/codegym/aurora/entity/ProductImage.java (hunk 6 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/entity/ProductImage.java]: FIXED

File: src/main/java/com/codegym/aurora/entity/SubCategory.java (hunk 7 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/entity/SubCategory.java]: FIXED

File: src/main/java/com/codegym/aurora/payload/response/ProductDetailResponseDTO.java (hunk 8 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/payload/response/ProductDetailResponseDTO.java:22]: FIXED

File: src/main/java/com/codegym/aurora/payload/response/ProductImageResponseDTO.java (hunk 9 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/payload/response/ProductImageResponseDTO.java]: FIXED

File: src/main/java/com/codegym/aurora/repository/UserRepository.java (hunk 10 above)
Comment (DUYNGOC97) [reply src/main/java/com/codegym/aurora/repository/UserRepository.java]: FIXED

**Comment by DUYNGOC97** [comment]
FIXED

**Pull request closed without merging**